src/instructlab/sdg/pipeline.py

```diff
@@ -78,6 +84,33 @@ def batching_enabled(self) -> bool:
         """
         return self.batch_size > 0 and self.batch_num_workers != 1
 
+    @property  # type: ignore
+    def client(self):
+        return self.clients.get(self.DEFAULT_CLIENT_KEY, None)
+
+    @client.setter
+    def client(self, value):
+        if isinstance(value, property):
+            # No default value
+            value = None
+        self.clients[self.DEFAULT_CLIENT_KEY] = value
+
+    @property  # type: ignore
+    def clients(self):
+        if self._clients is None:
+            self._clients = {}
+        return self._clients
+
+    @clients.setter
+    def clients(self, value):
+        if isinstance(value, property):
+            # Empty hash default value
+            value = {}
+        if value:
+            # Only set _clients if passed in a value, so we don't
+            # override it with the default of None from the @dataclass
+            self._clients = value
+
```

Comment:
Something like this could be added to make sure 'default' key is explicitly mentioned, so the blocks know how to behave during fallback path (?). Just a suggestion, please feel free to disregard or consider alternate implementations for the same.

```suggestion
    def __post_init__(self):
        if self.clients is not None and "default" not in self.clients:
            raise ValueError("PipelineContext requires a 'default' client in the clients dictionary")
```